flake.nix

```diff
@@ -33,12 +36,37 @@
           inherit system;
           overlays = [ overlay ];
         };
-      in
-      {
-        devShells.default = with pkgs; mkShell {
-          buildInputs = [
-            polykey-cli
-          ];
+      in rec {
+        devShells = {
+          # testing polykey-cli
+          polykey-cli = with pkgs; mkShell {
+            buildInputs = [
+              polykey-cli
+            ];
+          };
+          
+          # testing ignoreFromStore
+          gitignore = with pkgs; mkShell {
+            buildInputs = [
+              packages.gitignore
+            ];
+            shellHook = ''
+              echo "Output after filtering files:"
+              ls result -a
+            '';
+          };
+        };
+
+        packages = {
+          gitignore = pkgs.stdenv.mkDerivation {
+            pname = "gitignore-test";
+            version = "1.0.0";
+            src = pkgs.gitignore ["LICENSE" "README.md" "modules"] ./.;
+            buildPhase = ''
+              mkdir -p $out
+              cp -r $src/* $out/
+            '';
+          };
```

Comment:
What is this??

Reply:
This was meant to be a way to easily test out the functionality of the gitignore.

As per https://github.com/MatrixAI/nixpkgs-matrix/pull/4#issuecomment-2292635796, I will remove all the devshells and irrelevant packages.